### PR TITLE
PUBDEV-8742: Using the model performance API to avoid unnecessary transforms in the scikit-learn wrapper

### PR DESCRIPTION
**Pull request by ak-gupta**
In this PR, @NikhilJArora and I have updated the estimator score support in the `scikit-learn` wrapper to avoid unnecessary transformations between `h2o.H2OFrame` and `numpy.ndarray`. For classification models, we copied `sklearn.base.ClassifierMixin` and `sklearn.base.RegressorMixin` to determine the default metrics.

https://h2oai.atlassian.net/browse/PUBDEV-8742

### CODE REVIEW

**Comment by michalkurka** [comment]
thank you for your contribution @ak-gupta and @NikhilJArora

**Comment by ak-gupta** [comment]
Hi there, is there any update on a timeline for when this PR will be reviewed?

**Comment by sebhrusen** [comment]
@ak-gupta sorry, I forgot you, reviewing it now

File: h2o-py/h2o/sklearn/wrapper.py [hunk 1]
```diff
@@ -773,15 +773,23 @@ def delegate_score(delegate):
...
-        elif self.is_regressor():
-            with Mixin(self, RegressorMixin) as delegate:
-                return delegate_score(delegate)
+        scoring_frame = X if y is None else X.concat(y)
+        if hasattr(self._estimator, 'model_performance') and self.is_classifier():
+            # sklearn default classification metric is accuracy
+            return self._estimator.model_performance(scoring_frame).accuracy()
+        elif hasattr(self._estimator, 'model_performance') and self.is_regressor():
+            # sklearn default regression metric is r2
+            return self._estimator.model_performance(scoring_frame).r2()
```
Comment (sebhrusen) [inline h2o-py/h2o/sklearn/wrapper.py:782]: I think I understand why you would want to optimize this (avoid extra conversion as mentioned in `delegate_score` docstring), however we should keep the semantic of score in any case, so please don't apply the optimization if `sample_weight` is not `None`.
Also, ideally, no `scoring_frame` generation/concatenation if we're not going to use `model_performance`.

**Review by sebhrusen — CHANGES_REQUESTED** [review]

File: h2o-py/h2o/sklearn/wrapper.py (hunk 1 above)
Comment (ak-gupta) [reply h2o-py/h2o/sklearn/wrapper.py]: Makes sense -- thanks for the review! I've pushed changes, let me know if I need to do anything else

**Review by sebhrusen — APPROVED** [review]
Looks good, thanks @ak-gupta 
Merging now and will monitor tests.

**Pull request merged**